pkg/openstack/loadbalancer_test.go

```diff
@@ -534,3 +535,173 @@ func Test_getListenerProtocol(t *testing.T) {
+				healthCheckNodePort: 8081,
+				enableMonitor:       true,
+			},
+			expectedLen: 2,
+			expectError: false,
+		},
+		{
+			name:  "Valid nodes, canUseHTTPMonitor=true",
+			nodes: []*corev1.Node{node1, node2},
+			port:  corev1.ServicePort{NodePort: 8080},
+			svcConf: &serviceConfig{
+				preferredIPFamily:   corev1.IPv4Protocol,
+				lbMemberSubnetID:    "subnet-12345-test",
+				healthCheckNodePort: 8081,
+				enableMonitor:       true,
+			},
+			expectedLen: 2,
+			expectError: false,
+		},
+		{
+			name:  "ErrNoAddressFound happens and a member is ignored",
+			nodes: []*corev1.Node{node1, node2},
+			port:  corev1.ServicePort{NodePort: 8080},
+			svcConf: &serviceConfig{
+				preferredIPFamily:   corev1.IPv4Protocol,
+				lbMemberSubnetID:    "subnet-12345-test",
+				healthCheckNodePort: 8081,
+				enableMonitor:       false,
+			},
+			expectedLen: 1,
+			expectError: true, // Expect an error
+		},
+		{
+			name:        "Valid nodes, NodePort != 0, canUseHTTPMonitor=false",
+			nodes:       []*corev1.Node{node1, node2},
+			port:        port,
+			svcConf:     svcConf,
+			expectedLen: 2,
+			expectError: false,
+		},
+		{
+			name:  "Valid nodes, NodePort != 0, canUseHTTPMonitor=true",
+			nodes: []*corev1.Node{node1, node2},
+			port:  corev1.ServicePort{NodePort: 8080},
+			svcConf: &serviceConfig{
+				preferredIPFamily:   corev1.IPv4Protocol,
+				lbMemberSubnetID:    "subnet-12345-test",
+				healthCheckNodePort: 8081,
+				enableMonitor:       true,
+			},
+			expectedLen: 2,
+			expectError: false, // No error expected
+		},
+		{
+			name:  "Invalid preferred IP family, fallback to default",
+			nodes: []*corev1.Node{node1, node2},
+			port:  port,
+			svcConf: &serviceConfig{
+				preferredIPFamily:   "invalid-family",
+				lbMemberSubnetID:    "subnet-12345-test",
+				healthCheckNodePort: 8081,
+			},
+			expectedLen: 2,
+			expectError: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			members, newMembers, err := lbaas.buildBatchUpdateMemberOpts(tc.port, tc.nodes, tc.svcConf)
+
+			if tc.expectError {
+				// Expect an error
+				if assert.EqualError(t, err, "Expected an error in test case: "+tc.name) {
+					assert.Nil(t, members, "Members should be nil when there's an error in test case: "+tc.name)
+					assert.Nil(t, newMembers, "NewMembers should be nil when there's an error in test case: "+tc.name)
+				}
+			} else {
+				// No error expected
+				if assert.NoError(t, err, "Expected no error in test case: "+tc.name) {
+					assert.Len(t, members, tc.expectedLen, "Unexpected number of members in test case: "+tc.name)
+
+					// Check for empty newMembers when the flag is set
+					if tc.checkNewMembersEmpty {
+						if assert.Len(t, newMembers, 0, "NewMembers should be empty when there's no error in test case: "+tc.name) {
+							assert.True(t, newMembers.Has(fmt.Sprintf("%s-%s-%d-%d", node1.Name, "192.168.1.1", 8080, 8081)))
```

Comment:
`fmt` is not imported in this file

Reply:
Also, I'd try something like this:

```suggestion
							assert.Contains(t, newMembers, fmt.Sprintf("%s-%s-%d-%d", node1.Name, "192.168.1.1", 8080, 8081))
```

Since the underlying type for `Set` [is a map](https://github.com/kubernetes/apimachinery/blob/1ca6c696c8e470d0745788dad2f69de813b2237d/pkg/util/sets/set.go#L24), this has a chance of working.